src/controller/src/lib.rs

```diff
@@ -286,14 +324,55 @@ where
         } else {
             let state = Rc::new((t, token));
             for id in objects {
-                self.watch_sets
+                self.objects_to_unfulfilled_watch_sets
                     .entry(id)
                     .or_default()
                     .push(Rc::clone(&state));
             }
         }
     }
 
+    /// Process a pending response from the storage controller. If necessary,
+    /// return a higher-level response to our client.
+    ///
+    /// Precondition: The last readiness status generated must be `Readiness::Storage`.
```

Comment:
Is this really a necessary precondition? I don't think anything goes wrong if it's not fulfilled, the storage controller just won't have any work to do.